### PR TITLE
Make autodoc to skip sklearn inherited members

### LINKED ISSUE #1071: WARNING: undefined label: 'metadata_routing'
With recent (1.3+) versions of scikit-learn, building the documentation prints
> docstring of sklearn.utils._metadata_requests._MetadataRequester.get_metadata_routing:3: WARNING: undefined label: 'metadata_routing'
> docstring of sklearn.utils._metadata_requests.RequestMethod.__get__.<locals>.func:3: WARNING: undefined label: 'metadata_routing'

for Atol, MapperComplex, GraphInducedComplex and NerveComplex. New functions also get added to the documentation: `get_metadata_routing`, `set_fit_request`, `set_transform_request`. I don't know what criteria cause these to appear for some classes and not others.

Explanation is available at https://github.com/scikit-learn/scikit-learn/pull/26747

Some options for what we can do
1. ignore the warning
2. enable intersphinx
3. tell sphinx that we do not want those functions in the doc

We should probably check first what this new metadata_routing thing is about and whether it is relevant to us (if it is, we may want code changes, not just the doc).

### PR DESCRIPTION
**Pull request by VincentRouvreau**
Fix #1071 using option 3.

### CODE REVIEW

**Comment by VincentRouvreau** [comment]
In cover complexes, it removes members like:
* https://gudhi.inria.fr/python/latest/cover_complex_sklearn_isk_ref.html#gudhi.cover_complex.MapperComplex.get_metadata_routing
* https://gudhi.inria.fr/python/latest/cover_complex_sklearn_isk_ref.html#gudhi.cover_complex.MapperComplex.set_fit_request
* https://gudhi.inria.fr/python/latest/cover_complex_sklearn_isk_ref.html#gudhi.cover_complex.MapperComplex.set_params

In vector methods, it removes:
* https://gudhi.inria.fr/python/latest/representations.html#gudhi.representations.vector_methods.Atol.set_fit_request
* https://gudhi.inria.fr/python/latest/representations.html#gudhi.representations.vector_methods.Atol.set_transform_request

**Review by mglisse — APPROVED** [review]
